debug/gdbserver.py

```diff
@@ -717,21 +736,27 @@ def set_manual_trigger(self, tdata1, tdata2):
 
             tdata2_rb = self.gdb.p("$tdata2")
             tdata1_rb = self.gdb.p("$tdata1")
-            if tdata1_rb == tdata1 and tdata2_rb == tdata2:
-                return tselect
 
-            type_rb = tdata1_rb & MCONTROL_TYPE(self.hart.xlen)
-            type_none = set_field(0, MCONTROL_TYPE(self.hart.xlen),
-                                  MCONTROL_TYPE_NONE)
-            if type_rb == type_none:
+            type_rb = tdata1_rb & TDATA1_TYPE(self.hart.xlen)
+            type_match = set_field(0, TDATA1_TYPE(self.hart.xlen),
+                                   TDATA1_TYPE_MATCH)
+            type_none = set_field(0, TDATA1_TYPE(self.hart.xlen),
+                                  TDATA1_TYPE_NONE)
+            if type_rb == type_match:
+                maskmax_rb = tdata1_rb & MCONTROL_MASKMAX(self.hart.xlen)
```

Comment:
I'd sugest passing the mask to use before the comparison of `tdata1` and `tdata1_rb` as an argument of `set_manual_trigger()`